### PR TITLE
fix the timing issues

### PR DESCRIPTION
**Pull request by dylanb**
Here I fixed the issues with waiting for the page to actually load and then a bunch of promises stuff

### CODE REVIEW

**Comment by marcysutton** [comment]
Wow that got a lot more complex. Describing this as a screencast will be a promises demo as much as a testing demo. :raised_hands:

File: spec/test.js [hunk 1]
```diff
@@ -1,43 +1,55 @@
...
+            return element.getAttribute('class').then(function (className) {
+                expect(className.indexOf('home')).toBe(0);
+            });
+        }).then(function () {
+            AxeBuilder(browser)
+                .analyze(function(results) {
+                    expect(results.violations.length).toBe(0);
+                    done();
+                })
+        }).catch(function () {
```
Comment (marcysutton) [inline spec/test.js]: `catch` throws an error for me: 
`TypeError: driver.findElement(...).then(...).then(...).catch is not a function`

Removing that block fixes the error. Do we need it?

File: spec/test.js [hunk 2]
```diff
@@ -1,43 +1,55 @@
...
+            expect(false).toBe(true);
```
Comment (marcysutton) [inline spec/test.js]: What is this assertion doing?

File: spec/test.js [hunk 3]
```diff
@@ -1,43 +1,55 @@
...
 describe('Selenium-aXe Tutorial', function() {
 
     // Open the Deque website in the browser before each test is run
     beforeEach(function(done) {
-        this.driver = new selenium.Builder()
-            .forBrowser('firefox')
-            .build();
+        driver = new selenium.Builder()
+                .forBrowser('chrome')
+                .usingServer('http://localhost:4444/wd/hub');
```
Comment (marcysutton) [inline spec/test.js:14]: Does this assume you have started Selenium standalone in another tab? If so we should update the README (I have documented what I did to get the server running)

**Comment by dylanb** [comment]
You could if you make sure that the selector never fails, otherwise it will throw an exception.

**Comment by marcysutton** [comment]
Ok I'm going to close this PR, but I may reuse some of the concepts. Thanks @dylanb!

**Pull request closed without merging**